### PR TITLE
Document tables, and special parameters

### PR DESCRIPTION
**Pull request by sunfishcode**
This adds documentation to ir.rst for the newly added table feature, and for special parameters.

### CODE REVIEW

File: docs/ir.rst [hunk 1]
```diff
@@ -400,7 +400,7 @@ convention:
     retlist      : paramlist
     param        : type [paramext] [paramspecial]
     paramext     : "uext" | "sext"
-    paramspecial : "sret" | "link" | "fp" | "csr" | "vmctx"
+    paramspecial : "sret" | "link" | "fp" | "csr" | "vmctx" | "sigid" | "stack_limit"
```
Comment (bnjbvr) [inline docs/ir.rst]: Is it expected that the new `sigid` special parameter is added to the list, but never used in any definition? I'd expect to see it shown in the definition of a call instruction which can call into a table_addr.

Comment (sunfishcode) [reply docs/ir.rst]: I'm unclear on what you mean here. Are you looking for an example of how to use `sigid`?

Comment (bnjbvr) [reply docs/ir.rst]: Yes indeed, thanks for checking; other special parameters I've checked in the docs have a usage example, this one does not.

Comment (sunfishcode) [reply docs/ir.rst]: Ok, I opened https://github.com/CraneStation/cranelift/issues/497 to track the lack of an example for how to use `sigid`.

**Comment by sunfishcode** [comment]
Merging this, as it adds useful documentation, though of course there's more to do :-).

**Pull request merged**